pkg/engine/logging/default_logger.go

```diff
@@ -0,0 +1,43 @@
+package logging
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type defaultLogger struct {
+	entries []string
+}
+
+type logEventWrapper struct {
+	EventName string
+	Data      interface{}
+}
+
+func InitDefaultLogger() {
+	Singleton = &defaultLogger{entries: make([]string, 0, 10)}
+}
```

Comment:
I think it makes more sense to change this to the following and push `Singleton` assignment to the caller (since this way they also have full access to the specific logger type if there are any extra methods on it that are not defined in the `Logger` interface):
```go
func NewDefaultLogger() *DefaultLogger {
  return &defaultLogger{entries: make([]string, 0, 10)}
}